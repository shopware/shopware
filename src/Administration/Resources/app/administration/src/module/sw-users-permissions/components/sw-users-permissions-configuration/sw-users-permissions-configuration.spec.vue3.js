/**
 * @package system-settings
 */
import { mount } from '@vue/test-utils_v3';

async function createWrapper() {
    return mount(await wrapTestComponent('sw-users-permissions-configuration', {
        sync: true,
    }), {
        global: {
            stubs: {
                'sw-system-config': true,
            },
        },
    });
}

describe('module/sw-users-permissions/components/sw-users-permissions-configuration', () => {
    let wrapper;

    beforeEach(async () => {
        wrapper = await createWrapper();
    });

    it('should be a Vue.js component', () => {
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should emit the event correctly', () => {
        wrapper.vm.onChangeLoading(true);
        expect(wrapper.emitted('loading-change')).toBeTruthy();
    });
});
