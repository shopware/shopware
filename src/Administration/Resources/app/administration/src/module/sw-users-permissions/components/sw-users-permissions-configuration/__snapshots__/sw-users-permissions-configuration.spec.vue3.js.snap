// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module/sw-users-permissions/components/sw-users-permissions-configuration should be a Vue.js component 1`] = `
<sw-system-config-stub
  class="sw-users-permissions-configuration is--disabled"
  domain="core.userPermission"
/>
`;
